docker-compose.yaml

```diff
@@ -0,0 +1,42 @@
+version: "3.0"
+
+services:
+  server:
+    build:
+      context: .
+    restart: always
+    ports:
+      - "7171:7171"
+      - "7172:7172"
+    depends_on:
+      - db
+    networks:
+      - db-network
+  db:
+    image: mysql
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: <your_root_password>
+      MYSQL_USER: forgottenserver
+      MYSQL_PASSWORD: <your_db_password>
```

Comment:
Instead of that, try running both containers in the same network namespace and run a script that allows no password from `localhost` for the user.

Alternatively, try sharing a unix socket and using https://mariadb.com/kb/en/authentication-plugin-unix-socket/

Reply:
I'd recommend at least putting the variables in a separate file that can be in gitignore

```
    env_file:
      - db.env
```

Reply:
@DSpeichert that's not trivial with Docker, and I think he wants the password to be used in this "adminer" container below - I didn't even Google it, but it looks like phpMyAdmin.

Reply:
Sharing unix socket may not be easy due to uid mapping but sharing network namespaces is very easy with docker-compose:

```
...
services:
  db:
    ...
  pma:
    network_mode: "service:db"
    ...
...
```